src/test/scala/beam/agentsim/infrastructure/geozone/GeoZoneSpec.scala

```diff
@@ -0,0 +1,48 @@
+package beam.agentsim.infrastructure.geozone
+
+import java.io.FileInputStream
+import java.nio.file.{Files, Path, Paths}
+
+import beam.utils.FileUtils
+import org.apache.commons.codec.digest.DigestUtils
+import org.scalatest.{Matchers, WordSpec}
+
+class GeoZoneSpec extends WordSpec with Matchers {
+
+  "GeoZone combined with TopDownEqualDemandsGeoZoneHexGenerator and GeoZoneUtil" should {
+
+    "parse coordinates from austin and properly generate already tested GIS file" in {
+      FileUtils.usingTemporaryDirectory { tmpFolder =>
+        val csvPath = Paths.get("test", "input", "geozone", "austin.csv")
+        val wgsCoordinates = GeoZoneUtil.readWgsCoordinatesFromCsv(csvPath)
+        val summary = new GeoZone(wgsCoordinates).includeBoundBoxPoints
```

Comment:
Why don't just verify summary with what are you expecting, like H3Index -> Number of points there? Shape is about the geometry, but what you probably need to verify is that the points are distributes proportionally.

Reply:
It is a good point. For saving time I created this as an integration test that cover 3 components: `GeoZone`, `TopDownEqualDemandsGeoZoneHexGenerator` and `GeoZoneUtil`. Following your suggestion and making consistency I have to test each component independently and it will be split in 4 tests. It will take some time but is feasible. If you think is needed I can work on that.